core/src/main/java/hudson/node_monitors/TemporarySpaceMonitor.java

```diff
@@ -52,7 +54,8 @@ public TemporarySpaceMonitor(String freeSpaceThreshold) throws ParseException {
     public TemporarySpaceMonitor() {}
 
     public DiskSpace getFreeSpace(Computer c) {
-        return DESCRIPTOR.get(c);
+        DiskSpaceMonitorDescriptor descriptor = (DiskSpaceMonitorDescriptor) Jenkins.get().getDescriptor(TemporarySpaceMonitor.class);
```

Comment:
can't we use? `ExtensionList.lookupSingleton(DiskSpace.class)`

Reply:
Answer: No.

```
java.lang.IllegalStateException: Expected 1 instance of hudson.node_monitors.DiskSpaceMonitorDescriptor$DiskSpace but got 0
        at hudson.ExtensionList.lookupSingleton(ExtensionList.java:454)
        at hudson.node_monitors.TemporarySpaceMonitor.getFreeSpace(TemporarySpaceMonitor.java:65)
        at hudson.node_monitors.TemporarySpaceMonitor.getColumnCaption(TemporarySpaceMonitor.java:73)
        at hudson.model.ComputerSet.doUpdateNow(ComputerSet.java:234)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
        at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:397)
        at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:409)
        at org.kohsuke.stapler.interceptor.RequirePOST$Processor.invoke(RequirePOST.java:78)
        at org.kohsuke.stapler.PreInvokeInterceptedFunction.invoke(PreInvokeInterceptedFunction.java:26)
        at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:207)
        at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:140)
        at org.kohsuke.stapler.MetaClass$11.doDispatch(MetaClass.java:558)
        at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:59)
        at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:766)
```

Reverting the commit for this non-suggestion.